example/ornstein_uhlenbeck/CMakeLists.txt

```diff
@@ -6,6 +6,10 @@ make_catalogue_standalone(
     CXX_FLAGS_TARGET ${ARB_CXX_FLAGS_TARGET_FULL}
     VERBOSE ON)
 
+if(ARB_WITH_NVCC)
```

Comment:
There is also `ARB_WITH_CUDA_CLANG` and `ARB_WITH_HIP_CLANG` to consider.

Reply:
maybe, but then the tests should also handle this, and they do not - what is the difference?

Reply:
For HIP, we need to define `ARB_HIP` and `ARB_CUDA` for the other two. See https://github.com/arbor-sim/arbor/blob/master/CMakeLists.txt#L369

Reply:
yes, you are right. Changes:
- use `make_catalogue` instead of `make_catalogue_standalone`
- link to `arbor-private-deps`
- mark all files as c++ sources when using `cuda-clang` or `hip-clang`